src/app/component/create-loan/create-loan-form.tsx

```diff
@@ -42,10 +84,41 @@ const CreateLoanForm = () => {
     return <div>none</div>;
   };
 
-  const handleSelectChange = (value: string) => {
-    setSelectedValue(value);
+  const handleChange = (e: ChangeEvent<HTMLInputElement>): void => {
+    const { name, value } = e.target;
+    setFormValues((prevValues) => ({
+      ...prevValues,
+      [name]: value,
+    }));
   };
 
+  async function handleCreateLoanSubmit() {
+    await createNewLoan(formValues);
+    setShowSummary((prevState) => !prevState);
+  }
+
+  const { data: hash, writeContract } = useWriteContract();
+
+  async function submit(e: React.FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+    writeContract({
+      address: userWalletAddress,
+      abi,
+      functionName: "createLoanPosition",
+      args: [
+        address,
+        formValues?.loanAmount,
+        formValues?.loanAmountToken,
+        formValues?.collateralAmount,
+        formValues?.collateralAmountToken,
+        formValues?.loanPeriod,
+        formValues?.interestRate,
+      ],
+    });
```

Comment:
We don't call to any web3 or API directly from the section components.

Lets wrap this write contract into their own layer

Reply:
noted